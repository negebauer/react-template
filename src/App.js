import React, { Component } from "react"
import PropTypes from "prop-types"
import { Provider, connect } from "react-redux"
import { ConnectedRouter } from "react-router-redux"

import { devlog } from "./utils/log"
import Nav from "./Nav"
import { hydrate } from "./redux/modules/hydratation"

const mapStateToProps = state => ({
  hydratation: state.hydratation,
})

const mapDispatchToProps = {
  hydrate,
}

export class App extends Component {
  static propTypes = {
    store: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    hydratation: PropTypes.object.isRequired,
    hydrate: PropTypes.func.isRequired,
    options: PropTypes.object,
  }

  componentWillMount() {
    const { store, hydrate, options } = this.props
    hydrate(store, options.hydratation)
  }

  render() {
    devlog("App", this.state, this.props)
    if (!this.props.hydratation.done) {
      return null
    }
    return (
      <Provider store={this.props.store}>
        <ConnectedRouter history={this.props.history}>
          <Nav />
        </ConnectedRouter>
      </Provider>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
