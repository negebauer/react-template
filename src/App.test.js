// Maybe check
// https://github.com/Gethyl/ReactReduxTestingUsingJestEnzyme/blob/master/__test__/Home.spec.js

import React from "react"
import createHistory from "history/createBrowserHistory"
import configureStore from "./redux/store"
import { shallow, mount, render } from "enzyme"
import toJson from "enzyme-to-json"
import _ from "lodash/fp/object"
import { App } from "./App"

let history
let initialState
let store
let options
let defaultProps
let hydrate

beforeEach(() => {
  history = createHistory()
  initialState = { hydratation: { done: true } }
  store = configureStore(initialState, history)
  options = { hydratation: { blacklist: ["hydratation", "router"] } }
  hydrate = () => {}
  defaultProps = { history, store, options, hydrate, ...initialState }
})

it("shallow renders without crashing", () => {
  shallow(<App {...defaultProps} />)
})

it("mount renders without crashing", () => {
  mount(<App {...defaultProps} />)
})

// it("render renders without crashing", () => {
//   render(<App {...defaultProps} />)
// })

// it("matches snapshot", () => {
//   const wrapper = mount(<App {...defaultProps} />)
//   expect(toJson(wrapper)).toMatchSnapshot()
// })

// it("renders children", () => {
// const wrapper = mount(
//   <App store={store} options={options} history={history} />
// )
// expect(wrapper.children().exists()).toBeTruthy()
// })

it("renders null when no hydratation", () => {
  defaultProps.hydratation.done = false
  const wrapper = mount(<App {...defaultProps} />)
  expect(wrapper.children().exists()).toBeFalsy()
})
