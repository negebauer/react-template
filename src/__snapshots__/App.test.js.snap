// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<App
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  hydratation={
    Object {
      "done": true,
    }
  }
  hydrate={[Function]}
  options={
    Object {
      "hydratation": Object {
        "blacklist": Array [
          "hydratation",
          "router",
        ],
      },
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ConnectedRouter
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <withRouter(Connect(Navigator))>
          <Route
            render={[Function]}
          >
            <Connect(Navigator)
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Navigator
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <styled.div>
                  <div
                    className="sc-gzVnrw kugQOZ"
                  >
                    <styled.div>
                      <div
                        className="sc-htoDjs ePHjYV"
                      >
                        <Styled(styled.div)>
                          <styled.div
                            className="sc-iwsKbI jQhHGA"
                          >
                            <div
                              className="sc-iwsKbI jQhHGA sc-dnqmqq cbdZBN"
                            >
                              <NavLink
                                exact={true}
                                ignore={true}
                                label="React Template"
                                to="/"
                              >
                                <Route
                                  exact={true}
                                  path="/"
                                >
                                  <Styled(Link)
                                    className=""
                                    to="/"
                                  >
                                    <Link
                                      className="sc-gqjmRU eNtcLP"
                                      replace={false}
                                      to="/"
                                    >
                                      <a
                                        className="sc-gqjmRU eNtcLP"
                                        href="/"
                                        onClick={[Function]}
                                      >
                                        React Template
                                      </a>
                                    </Link>
                                  </Styled(Link)>
                                </Route>
                              </NavLink>
                              <NavLink
                                label="Other 1"
                                to="/other1"
                              >
                                <Route
                                  path="/other1"
                                >
                                  <Styled(Link)
                                    className=""
                                    to="/other1"
                                  >
                                    <Link
                                      className="sc-gqjmRU eNtcLP"
                                      replace={false}
                                      to="/other1"
                                    >
                                      <a
                                        className="sc-gqjmRU eNtcLP"
                                        href="/other1"
                                        onClick={[Function]}
                                      >
                                        Other 1
                                      </a>
                                    </Link>
                                  </Styled(Link)>
                                </Route>
                              </NavLink>
                              <NavLink
                                label="Other 2"
                                to="/other2"
                              >
                                <Route
                                  path="/other2"
                                >
                                  <Styled(Link)
                                    className=""
                                    to="/other2"
                                  >
                                    <Link
                                      className="sc-gqjmRU eNtcLP"
                                      replace={false}
                                      to="/other2"
                                    >
                                      <a
                                        className="sc-gqjmRU eNtcLP"
                                        href="/other2"
                                        onClick={[Function]}
                                      >
                                        Other 2
                                      </a>
                                    </Link>
                                  </Styled(Link)>
                                </Route>
                              </NavLink>
                            </div>
                          </styled.div>
                        </Styled(styled.div)>
                        <Styled(styled.div)>
                          <styled.div
                            className="sc-gZMcBi giNnHm"
                          >
                            <div
                              className="sc-gZMcBi giNnHm sc-dnqmqq cbdZBN"
                            >
                              <NavLink
                                label="Accesos"
                                to="/go"
                              >
                                <Route
                                  path="/go"
                                >
                                  <Styled(Link)
                                    className=""
                                    to="/go"
                                  >
                                    <Link
                                      className="sc-gqjmRU eNtcLP"
                                      replace={false}
                                      to="/go"
                                    >
                                      <a
                                        className="sc-gqjmRU eNtcLP"
                                        href="/go"
                                        onClick={[Function]}
                                      >
                                        Accesos
                                      </a>
                                    </Link>
                                  </Styled(Link)>
                                </Route>
                              </NavLink>
                            </div>
                          </styled.div>
                        </Styled(styled.div)>
                      </div>
                    </styled.div>
                    <Switch>
                      <Route
                        component={[Function]}
                        computedMatch={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                        exact={true}
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        path="/"
                      >
                        <styled.div>
                          <div
                            className="sc-VigVT lnHQkM"
                          >
                            <HelmetWrapper
                              defer={true}
                              encodeSpecialCharacters={true}
                            >
                              <SideEffect(NullComponent)
                                defer={true}
                                encodeSpecialCharacters={true}
                                title="React Template"
                                titleAttributes={Object {}}
                              >
                                <NullComponent
                                  defer={true}
                                  encodeSpecialCharacters={true}
                                  title="React Template"
                                  titleAttributes={Object {}}
                                />
                              </SideEffect(NullComponent)>
                            </HelmetWrapper>
                            <Route
                              component={[Function]}
                              computedMatch={
                                Object {
                                  "isExact": true,
                                  "params": Object {},
                                  "path": "/",
                                  "url": "/",
                                }
                              }
                              exact={true}
                              location={
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                }
                              }
                              path="/"
                            >
                              <Connect(Home)
                                history={
                                  Object {
                                    "action": "POP",
                                    "block": [Function],
                                    "createHref": [Function],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
                                }
                                location={
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  }
                                }
                                match={
                                  Object {
                                    "isExact": true,
                                    "params": Object {},
                                    "path": "/",
                                    "url": "/",
                                  }
                                }
                              >
                                <Home
                                  history={
                                    Object {
                                      "action": "POP",
                                      "block": [Function],
                                      "createHref": [Function],
                                      "go": [Function],
                                      "goBack": [Function],
                                      "goForward": [Function],
                                      "length": 1,
                                      "listen": [Function],
                                      "location": Object {
                                        "hash": "",
                                        "pathname": "/",
                                        "search": "",
                                        "state": undefined,
                                      },
                                      "push": [Function],
                                      "replace": [Function],
                                    }
                                  }
                                  location={
                                    Object {
                                      "hash": "",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    }
                                  }
                                  match={
                                    Object {
                                      "isExact": true,
                                      "params": Object {},
                                      "path": "/",
                                      "url": "/",
                                    }
                                  }
                                >
                                  <Styled(styled.div)>
                                    <styled.div
                                      className="sc-htpNat hTNTjW"
                                    >
                                      <div
                                        className="sc-htpNat hTNTjW sc-bdVaJa cljovs"
                                      >
                                        <h1>
                                          React Template
                                        </h1>
                                      </div>
                                    </styled.div>
                                  </Styled(styled.div)>
                                </Home>
                              </Connect(Home)>
                            </Route>
                          </div>
                        </styled.div>
                      </Route>
                    </Switch>
                    <styled.div>
                      <div
                        className="sc-jTzLTM hrEXlU"
                      >
                        <Styled(styled.div)>
                          <styled.div
                            className="sc-jzJRlG enxKql"
                          >
                            <div
                              className="sc-jzJRlG enxKql sc-fjdhpX gIIVcx"
                            >
                              <FooterLink
                                label="Google"
                                to="https://google.com"
                              >
                                <styled.a
                                  href="https://google.com"
                                  target="_blank"
                                >
                                  <a
                                    className="sc-kAzzGY hWxvCe"
                                    href="https://google.com"
                                    target="_blank"
                                  >
                                    Google
                                  </a>
                                </styled.a>
                              </FooterLink>
                            </div>
                          </styled.div>
                        </Styled(styled.div)>
                        <Styled(styled.div)>
                          <styled.div
                            className="sc-cSHVUG CAuLq"
                          >
                            <div
                              className="sc-cSHVUG CAuLq sc-fjdhpX gIIVcx"
                            >
                              <FooterLink
                                label="Github"
                                to="https://github.com/negebauer/react-template"
                              >
                                <styled.a
                                  href="https://github.com/negebauer/react-template"
                                  target="_blank"
                                >
                                  <a
                                    className="sc-kAzzGY hWxvCe"
                                    href="https://github.com/negebauer/react-template"
                                    target="_blank"
                                  >
                                    Github
                                  </a>
                                </styled.a>
                              </FooterLink>
                            </div>
                          </styled.div>
                        </Styled(styled.div)>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </Navigator>
            </Connect(Navigator)>
          </Route>
        </withRouter(Connect(Navigator))>
      </Router>
    </ConnectedRouter>
  </Provider>
</App>
`;
